recipes/naive-tsearch/all/conanfile.py

```diff
@@ -0,0 +1,78 @@
+from conans import CMake, ConanFile, tools
+import os
+
+
+class NaiveTsearchConan(ConanFile):
+    name = "naive-tsearch"
+    description = "A simple tsearch() implementation for platforms without one"
+    topics = ("conan", "naive-tsearch", "tsearch", "search", "tree", "msvc")
+    license = "MIT"
+    homepage = "https://github.com/kulp/naive-tsearch"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+    exports_sources = ["CMakeLists.txt"]
+    options = {
+        "fPIC": [True, False],
+        "header_only": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+        "header_only": True,
+    }
+    generators = "cmake"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["NAIVE_TSEARCH_INSTALL"] = True
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def package_id(self):
+        del self.info.options.header_only
+
+    def package_info(self):
+        if self.options.header_only:
+            self.cpp_info.components["header_only"].libs = []
+            self.cpp_info.components["header_only"].libdirs = []
```

Comment:
Is this required?

Reply:
Well, I don't  know.
But I saw that the libdir was included in the `DYLD_LIBRARY_PATH` of https://github.com/conan-io/conan-center-index/pull/2342#issuecomment-666815330, even though the library is header_only.

Reply:
I thought about it a bit more.

I think it is required.
The conan default for libdirs is `['lib']`.
This recipe has always a library in the `lib` folder, even when `header_only=True`.
A *normal* header package does not have a `lib` folder and so `lib_paths` will reduce to the empty list.
So to avoid using the `lib` folder, I explicitly set it to the empty list.

Reply:
Ahh that makes sense, perhaps copy when using header only instead of install?

Reply:
The library is always built. It's just not in the library path when header_only is true.

Reply:
Or maybe pass from the build with header only 🤔 but make might fail from missing targets

Reply:
Indeed, install installs headers + libraries. So the library will be built always.
I like how there is now one package, whatever header_only's value is.
An alternative would be to remove the `lib` folder after installation, when header_only is true.